app/src/main/java/org/simple/clinic/patient/PatientFuzzySearch.kt

```diff
@@ -14,20 +14,36 @@ class PatientFuzzySearch {
   interface PatientFuzzySearchDao {
 
     fun searchForPatientsWithNameLikeAndAgeWithin(query: String, dobUpperBound: String, dobLowerBound: String): Single<List<PatientSearchResult>>
+
+    fun searchForPatientsWithNameLike(query: String): Single<List<PatientSearchResult>>
   }
 
   class PatientFuzzySearchDaoImpl(
       private val sqLiteOpenHelper: SupportSQLiteOpenHelper,
       private val patientSearchDao: PatientSearchResult.RoomDao
   ) : PatientFuzzySearchDao {
 
-    override fun searchForPatientsWithNameLikeAndAgeWithin(query: String, dobUpperBound: String, dobLowerBound: String) =
-        patientUuidsMatching(query).flatMap { uuidsSortedByScore ->
-          val uuids = uuidsSortedByScore.map { it.uuid }
-          patientSearchDao
-              .searchByIds(uuids, dobUpperBound, dobLowerBound, PatientStatus.ACTIVE)
-              .compose(sortPatientSearchResultsByScore(uuidsSortedByScore))
-        }!!
+    override fun searchForPatientsWithNameLikeAndAgeWithin(
+        query: String,
+        dobUpperBound: String,
+        dobLowerBound: String
+    ): Single<List<PatientSearchResult>> {
+      return patientUuidsMatching(query).flatMap { uuidsSortedByScore ->
```

Comment:
Shouldn't we be using a `switchMap` here?

Reply:
We could, but that would mean that we would have to convert the `patientUuidsMatching` method to a room Dao method instead of querying the data ourselves.

Given that is an unused data flow (since searchV2 flag has been set), I don't think we should spend time on doing this now. Maybe we can revisit this if we ever decide to implement the new fuzzy search in SQLite again.

Reply:
Got it